src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7690,9 +7707,10 @@ private bool IsPossibleLocalDeclarationStatement(bool isGlobalScriptLevel)
 
             tk = this.CurrentToken.ContextualKind;
 
-            var isPossibleAttributeOrModifier = (IsAdditionalLocalFunctionModifier(tk) || tk == SyntaxKind.OpenBracketToken)
-                && (tk is not (SyntaxKind.AsyncKeyword or SyntaxKind.ScopedKeyword) || ShouldContextualKeywordBeTreatedAsModifier(parsingStatementNotDeclaration: true));
-            if (isPossibleAttributeOrModifier)
+            var isPossibleModifier = IsAdditionalLocalFunctionModifier(tk)
+                && (tk is not (SyntaxKind.AsyncKeyword or SyntaxKind.ScopedKeyword)
+                || ShouldContextualKeywordBeTreatedAsModifier(parsingStatementNotDeclaration: true));
```

Comment:
we don't need to check for attributes here since attributes have already been parsed out before this.